### DISCUSSION TITLE
Xmpp.js in react native

### DISCUSSION THREAD

**Discussion by Saitejakolan9** [xmppjs/xmpp.js#953] [Q&A]
Can some let me know the steps to use xmpp.js in react Native?
Like installation.. (npm command) etc.

**Comment by Bogdan14793478** [comment]
i write exemple of code, but no connect - Error connect
 import React, { useState, useEffect } from 'react'
import { Text, TouchableHighlight, View } from 'react-native'
import { xml, client } from '@xmpp/client'
const debug = require('@xmpp/debug')

const XmppChat = () => {
  const [output, setOutput] = useState('')
  const [xmppClient, setXmppClient] = useState(null)

 

  const XMPPServerOptions = {
    service: 'ws://xmpp-staging.******.com:5222/ws',
    domain: 'xmpp-staging.******.com',
    resource: 'example',
    username: '*****',
    password: '70708397d9a456a4e9e8cb6*******4cfc82f0'
  }

  const connect = async () => {
    const xmpp = client(XMPPServerOptions)
    xmpp.jid = '****@xmpp-staging.*****.com' 
     setXmppClient(xmpp)

    debug(xmpp, true, '111debug connect')
    
    xmpp.on('online', address => {
      console.log('Connected as:', address.toString())
    })

    xmpp.on('error', error => {
      console.error('Error:', error)
    })

    xmpp.start().catch(error => {
      console.error('Error connecting to XMPP server:', error)
    })
   
    console.log(xmpp, 'xmpp1')
  }

  const sendMessage = () => {
    if (xmppClient) {
      const stanzaParams = {
        from: 'Bo....@xmpp.****.com',
        to: 'Tes...@xmpp.*****.com',
        type: 'chat',
        id: Math.floor(Math.random() * Math.floor(999999999))
      }
      const messageStanza = xmppClient.xmlElement('message', stanzaParams)
      messageStanza.c('body', { xmlns: 'jabber:client' }).t('Hello Amigo').up()
      xmppClient.send(messageStanza)
    }
  }

  const log = text => {
    console.log(text, 'text from XMPP log')
    setOutput(prevOutput => prevOutput + '\n' + text)
  }

  useEffect(() => {
    if (xmppClient) {
      xmppClient.on('connect', status => {
        xmppClient.jid = '*****@xmpp-staging.******.com'
        debug('status', status)
        log('CONNECTING')
      })
      xmppClient.on('online', jid => {
        xmppClient.jid = '****@xmpp-staging.******.com'

        log('ONLINE: ' + jid.toString())
      })
      xmppClient.on('stanza', stanza => {
        log('RECEIVED STANZA: ' + stanza.toString())
      })
    }
  }, [xmppClient])

  const onStartConnect = () => {
    connect()
  }

  return (
    <View>
      <Text>{output}</Text>
      <TouchableHighlight onPress={onStartConnect}>
        <Text>Connect</Text>
      </TouchableHighlight>
      <TouchableHighlight onPress={sendMessage}>
        <Text>Send Message</Text>
      </TouchableHighlight>
    </View>
  )
}

export default XmppChat
